### PR TITLE
Optimize script generation time

### PR DESCRIPTION
**Pull request by MortenTobiasNielsen**
I have left a few comments, to make it a little easier to understand the changes.

### CODE REVIEW

File: cMain.cpp [hunk 1]
```diff
@@ -3489,51 +3517,70 @@ void cMain::find_new_orientation() {
 
 bool cMain::find_building_for_script(int& row) {
 	for (int i = row - 1; i > -1; i--) {
-		building_x_cord = grid_tasks->GetCellValue(i, 1);
-		building_y_cord = grid_tasks->GetCellValue(i, 2);
```
Comment (MortenTobiasNielsen) [inline cMain.cpp:3493]: It turns out that it was primarily due to these lines being moved out of the if statements. I should have thought about that - this method is executed 1000's of times on script generation, so to get the x and y coordinates when they are not needed is just bad practice. :)

File: cMain.cpp [hunk 2]
```diff
@@ -3489,51 +3517,70 @@ void cMain::find_new_orientation() {
...
+		if (compare_task_strings(grid_tasks->GetCellValue(i, 0), struct_tasks_list.build)) {
```
Comment (MortenTobiasNielsen) [inline cMain.cpp:3520]: Apparently the string comparer isn't that efficient. This turns out to be much better)

File: utils.h [hunk 3]
```diff
@@ -785,7 +785,7 @@ struct {
 	std::string uranium_fuel_cell = "Uranium fuel cell";
 } struct_fuel_list;
 
-struct {
+static const struct {
```
Comment (MortenTobiasNielsen) [inline utils.h:788]: I don't think turning these into static const structs had much to do with the performance improvements, but might as well have them.

File: cMain.cpp (hunk 1 above)
Comment (theis999) [reply cMain.cpp:3493]: It makes sense that calls to the gui would be the culprit

**Review by theis999 — APPROVED** [review]
It looks reasonable.
If it continues being a problem then we could look at either multithreading the process or make a prepared output.

**Comment by MortenTobiasNielsen** [comment]
Yeah I was also thinking about multithreading, but it it can also get extremely convoluted.

**Pull request merged**